src/calendar/view/CalendarAgendaViewModel.ts

```diff
@@ -0,0 +1,20 @@
+import { CalendarEvent } from "../../api/entities/tutanota/TypeRefs.js"
+import { isAllDayEvent } from "../../api/common/utils/CommonCalendarUtils.js"
+
+export class CalendarAgendaViewModel {
+	earliestEventToShowTimeIndicator(events: readonly CalendarEvent[], date: Date): number | null {
+		// Get all events and indices, filtering out all-day events and events preceding the current time
+		let eventWithIndices = events
+			.map((event, index) => {
+				return { event, index }
+			})
+			.filter(({ event, index }) => !isAllDayEvent(event) && date < event.startTime)
+
+		// Remove events that have similar start dates but are lower on the list (have a higher index)
+		let eligibleIndices = eventWithIndices.filter(
```

Comment:
```suggestion
		const eligibleIndices = eventWithIndices.filter(
```

This does not seem to be mutated.